Config

evanH13 opened the pull request:
I still need to incorporate error-handling. I can also write more code that will update instead of overwriting values if args are passed via the CLI.

evanH13 committed:
add webserver, index, openbrowser

evanH13 committed:
write to configfile

evanH13 committed:
generate /authorize url and take user there

evanH13 committed:
rework for configparse

evanH13 committed:
rework CLI function, pass config file to main function, generate URL in main function

evanH13 committed:
web server class to handle auth code

evanH13 committed:
write config in main function

evanH13 committed:
made configfile global

evanH13 committed:
remove index.html, get auth code

evanH13 committed:
create server by parsing redirect_uri

evanH13 committed:
webserver now only handles request when authcode is not present

evanH13 committed:
woked in Wes' secure server code

evanH13 committed:
add index page and redirect to it

evanH13 committed:
get access and refresh tokens

evanH13 committed:
Clean up

evanH13 committed:
Assign access and refresh token values

evanH13 committed:
Prioritize CLI arugments, then configfile if specified. If configfile exists, only pass configfile param to use it

evanH13 committed:
add comments

evanH13 committed:
wrap configuration in it's own function

evanH13 committed:
create config module and rework to allow user input

evanH13 committed:
fixed doctstrings, added types, made class camelcase

evanH13 committed:
Prompts user for all input after running config arg with no subargs

evanH13 committed:
Fix if/else congif block, configfile optional

evanH13 committed:
add default configfile location

evanH13 committed:
write access and refresh tokens to configfile

evanH13 committed:
function to refresh access token with refresh token

evanH13 committed:
Config function now checks for refresh token, refresh access token if possible, otherwise auth modules are used to generate tokens and then all values written to configfile

evanH13 committed:
Update config.py

wtanner commented:
Tests are failing. I'll see if I can fix it up.

wtanner commented on spotifyutils/config.py:
```diff
@@ -0,0 +1,118 @@
+
+    config = configparser.ConfigParser()
+
+    if kwargs['configfile']:
```
This fails if there is no configuration file specified. That is not expected behavior.

wtanner commented on spotifyutils/config.py:
```diff
@@ -0,0 +1,118 @@
+
+    if kwargs['configfile']:
+        configfile = kwargs['configfile']
+        if kwargs['client_id'] or kwargs['client_secret'] or kwargs['redirect_uri']:
```
Same with these.

evanH13 commented:
Sounds good

wtanner committed:
Refactor the configuration functionality

1. Clean up config file parsing and writing
2. Add cryptography library dependency to setup.py
3. Fix test cases

wtanner committed:
fix config test case

wtanner committed:
specify the minimum python version as 3.6

wtanner commented:
> Sounds good
> 
> 

I made a pull request against this one for you to review, instead of a bunch of comments. When you are done reviewing we can merge that one into this one and then config is done.

evanH13 committed:
Merge pull request #12 from wtanner/refactor_config

Refactor configuration and fix tests
